sql2.txt

```diff
@@ -2,51 +2,84 @@
+    GROUP BY Products.ProductID, Products.ProductName
+    ORDER BY Products.ProductName ASC;
 
 
 2. Retrieve the customer name and OrderDetailsID for all the OrderDetails for each customer,
 ordered by customer name. Paste your SQL statement below. You will have to join the Customers
 table with the Orders table, and the Orders table with the OrderDetails table, 
 because the Orders table is the one that ties the Customer table to the OrderDetails table.
-
+SELECT Customers.CustomerName, OrderDetails.OrderDetailID
+FROM OrderDetails
+    JOIN Customers on Orders.CustomerID = Customers.CustomerID
+    JOIN Orders on OrderDetails.OrderID = Orders.OrderID
+    GROUP BY Customers.CustomerName, OrderDetails.OrderDetailID 
+    ORDER BY Customers.CustomerName ASC;
 
 3. Extend the previous query to join the Products table.  You are now joining the Customers, Orders,
 OrderDetails, and Products tables.  The result of your query should include the
 customer name, the product name, and the total quantity of that product ordered.  Note that a customer
 may have ordered the same product in several different orders.  So the total quantity is
 SUM(Quantity) AS TotalQuantity and you must GROUP BY CustomerID, ProductID.  Paste your
 SQL statement below.
-
-
+SELECT Customers.CustomerName, Products.ProductName, SUM(OrderDetails.Quantity) AS TotalQuantity
+FROM Customers
+    JOIN Orders ON Customers.CustomerID = Orders.CustomerID
+    JOIN OrderDetails ON Orders.OrderID = OrderDetails.OrderID
+    JOIN Products ON OrderDetails.ProductID = Products.ProductID
+GROUP BY Customers.CustomerID, Customers.CustomerName, Products.ProductID, Products.ProductName;
 
 4. Extend the previous query to give the total price as well as the customer name, product name, 
 and total quantity.  Here you will have
 (SUM(Quantity) * Price) AS TotalPrice.  Paste your SQL statement below.
-
+SELECT Customers.CustomerName, Products.ProductName, (SUM(OrderDetails.Quantity) * Price) AS TotalPrice
+FROM Customers
+    JOIN Orders ON Customers.CustomerID = Orders.CustomerID
+    JOIN OrderDetails ON Orders.OrderID = OrderDetails.OrderID
+    JOIN Products ON OrderDetails.ProductID = Products.ProductID
+GROUP BY Customers.CustomerID, Customers.CustomerName, Products.ProductID, Products.ProductName;
 
 5. Add a new Order to the Orders table.  You will have to specify the ID of an existing customer
 to satisfy the foreign key constraint.  You will also have to specify the ID of an existing
 employee.  Paste your SQL statement below.
-
+INSERT INTO Orders(OrderID, CustomerID, EmployeeID, OrderDate, ShipperID)
+VALUES (10247, 90, 5, "6/25/2024", 3);
 
 6. Add three OrderDetails.  These should specify the ID of the new Order you just created.  For each,
 you will have to specify the ID of an existing product.  Paste your SQL statement below.
+INSERT INTO OrderDetails(OrderDetailID, OrderID, ProductID, Quantity)
+VALUES (519, 10443, 29, 13), (520, 10443, 30, 14), (521, 10443, 31, 15);
```

Comment:
The instructions are to "specify the ID of the new Order you just created", but `10443` is the ID of the last order in the starter data, rather than of the row you just created. If you create an order with a hardcoded OrderID of `10247`, then you instead want:

```suggestion
INSERT INTO OrderDetails(OrderDetailID, OrderID, ProductID, Quantity)
VALUES (519, 10247, 29, 13), (520, 10247, 30, 14), (521, 10247, 31, 15);
```

And similar to my previous comment, you can let the database handle the creation of the OrderDetailID's like so:

```suggestion
INSERT INTO OrderDetails(OrderID, ProductID, Quantity)
VALUES (10443, 29, 13), (10443, 30, 14), (10443, 31, 15);
```